include/RAJA/pattern/params/reducer.hpp

```diff
@@ -15,46 +15,18 @@
 namespace RAJA
 {
 
-namespace expt
-{
-
-template<typename T>
-struct ValLoc {
-  using index_type = RAJA::Index_type;
-  using value_type = T;
-
-  RAJA_HOST_DEVICE ValLoc() {}
-  RAJA_HOST_DEVICE ValLoc(value_type v) : val(v) {}
-  RAJA_HOST_DEVICE ValLoc(value_type v, RAJA::Index_type l) : val(v), loc(l) {}
-
-  RAJA_HOST_DEVICE void min(value_type v, index_type l) { if (v < val) { val = v; loc = l; } }
-  RAJA_HOST_DEVICE void max(value_type v, index_type l) { if (v > val) { val = v; loc = l; } }
-
-  bool constexpr operator<(const ValLoc& rhs) const { return val < rhs.val; }
-  bool constexpr operator>(const ValLoc& rhs) const { return val > rhs.val; }
-
-  value_type getVal() {return val;}
-  RAJA::Index_type getLoc() {return loc;}
-
-private:
-  value_type val;
-  index_type loc = -1;
-};
-
-} //  namespace expt
-
 namespace operators
 {
 
-template <typename T>
-struct limits<RAJA::expt::ValLoc<T>> {
-  RAJA_INLINE RAJA_HOST_DEVICE static constexpr RAJA::expt::ValLoc<T> min()
+template <typename T, typename IndexType>
+struct limits<RAJA::expt::ValLoc<T, IndexType>> {
```

Comment:
Should ValLoc be moved into a more generic header?

Reply:
Maybe, but I will leave it here for now.